lib/reqTemplate.js

```diff
@@ -55,14 +54,45 @@ function compileTemplate(templateSpec, setValue, reqPart) {
     if (template[0] === '$') {
         template = 'rm' + template.slice(1);
     } else {
-        template = 'm.request.' + reqPart + '.' + template;
+        template = 'm.request.' + defaultLookupPath + '.' + template;
     }
-    template = template.replace(/\.([^.]*(?:-[^.]*)+)/g, function(all, paren) {
+    return template.replace(/\.([^.]*(?:-[^.]*)+)/g, function(all, paren) {
         return "['" + paren.replace(/'/g, "\\'") + "']";
     });
+}
+
+/**
+ * Compiles a template into a function
+ *
+ * @param templateSpec template string (e.g. '{$.request.body.a}')
+ * @param reqPart part of the request to lookup local properties in (e.g. headers, body etc.)
+ * @param setValue callback for a result of template evaluation
+ * @returns {[Function]} array of template resolvers for the provided template
+ */
+function compileTemplate(templateSpec, setValue, reqPart) {
+    var template;
+    var templatedString;
+    var prevIndex = 0;
+    var completeTemplate = /^\{([^}]+)}$/.exec(templateSpec);
+    if (completeTemplate && completeTemplate.length > 0) {
+        template = [['raw', prepareTemplateString(completeTemplate[1], reqPart)]];
+    } else {
+        template = [];
+        templateSpec.replace(/\{([^}]+)}/g, function(full, spec, index) {
```

Comment:
You aren't really replacing here. Instead, you should be able to use `.exec` like this (untested):

```javascript
var re = /\{([^}]+)}/g;
var template = [];
var i = 0;
var match;
do {
  match = re.exec(templateSpec);
  if (match) {
    if (match.index !== i) {
      template.push(templateSpec.substring(i, match.index);
    }
    template.push(['raw', prepareTemplateString(match[1], reqPart)]);
  } else if (i !== templateSpec.length) {
    template.push(templateSpec.substring(i, templateSpec.length);
  }
} while (match)
```